builds/misc/images.yaml

```diff
@@ -12,11 +12,10 @@ jobs:
 ################################################################################
     displayName: LinuxDotnet
     pool:
-      vmImage: ubuntu-18.04
+      name: $(pool.linux.name)
+      demands:
+        - ImageOverride -equals agent-aziotedge-ubuntu-18.04-docker
     steps:
-      - template: ../templates/install-dotnet2.yaml
-      - template: ../templates/install-dotnet3.yaml
-
       - bash: 'docker login $(registry.address) --username $(registry.user) --password $(registry.password)'
```

Comment:
Maybe this is a good time to introduce the Login Docker task to the E2E pipeline? 
https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/build/docker?view=azure-devops#login

I have a backlog task to basically address for all pipelines. Since this is more or less a brand new look for E2E, we might as well just address the issue now?

Reply:
The ultimate goal would be to remove the Docker secrets from the pipeline. The problem is that switching to the Docker task, by itself, doesn't accomplish that goal--at least not for the `images.yaml` pipeline. There's another step that needs to be taken: we need to update `buildManifest.sh` to not require username/password anymore. It uses them to do `docker login`, which is unnecessary in the pipeline case, so those script arguments should be optional.

If it was a simple switch to the Docker task I'd do it, but since there's more involved, let's keep this PR focused on the hosted pools.